mera/mera_pymer4.py

```diff
@@ -344,3 +253,172 @@ def run_mera(
+
+    # Check for nans
+    if cur_residual_df[im].isna().sum() > 0:
+        raise ValueError(f"NaNs found in IM {im}")
+
+    # Create and fit the model
+    if len(cur_residual_df) > 0:
+        # With site-term
+        if site_cname is not None:
+            cur_model = Lmer(
+                f"{im} ~ {'1' if assume_biased else '0'} + (1|{event_cname}) + (1|{site_cname})",
+                data=cur_residual_df,
+            )
+            cur_model.fit(summary=False)
+
+            # Get the site and event random effects
+            # (Ensure we extract the right dataframe with correct indexes)
+            event_index = (
+                0
+                if np.any(cur_model.ranef[0].index.isin(event_res_series.index))
+                else 1
+            )
+            site_index = (
+                1 if np.any(cur_model.ranef[1].index.isin(site_res_series.index)) else 0
+            )
+            event_re = cur_model.ranef[event_index].iloc[:, 0]
+            site_re = cur_model.ranef[site_index].iloc[:, 0]
+
+            # Get site term and standard deviation
+            site_res_series.loc[site_re.index] = site_re
+            bias_std_series.loc["phi_S2S"] = cur_model.ranef_var.loc[site_cname, "Std"]
+
+        # Without site-term
+        else:
+            cur_model = Lmer(
+                f"{im} ~ {'1' if assume_biased else '0'} + (1|{event_cname})",
+                data=cur_residual_df,
+            )
+            cur_model.fit(summary=False)
+
+            # Get the event term
+            # (Ensure we extract the right dataframe with correct indexes)
+            event_re = cur_model.ranef.iloc[:, 0]
+
+        # Get event and remaining terms
+        event_res_series.loc[event_re.index] = event_re
+        if mask is None:
+            rem_res_df[im] = cur_model.residuals
+            fit_series.values[:] = list(cur_model.fits)
+        else:
+            rem_res_df.loc[mask[im], im] = cur_model.residuals
+            fit_series.loc[mask[im]] = list(cur_model.fits)
+
+        # Get bias
+        if assume_biased:
+            bias_std_series.loc["bias"] = cur_model.coefs.iloc[0, 0]
+            bias_std_series.loc["bias_std_err"] = cur_model.coefs["SE"].iloc[0]
+
+        # Get standard deviations
+        bias_std_series.loc["tau"] = cur_model.ranef_var.loc[event_cname, "Std"]
+        bias_std_series.loc["phi_w"] = cur_model.ranef_var.loc["Residual", "Std"]
+
+        # set the index of ranef_df so it matches those of the output DataFrames
+        cur_model.ranef_df.set_index("grp", inplace=True)
+        cur_model.ranef_df.index.rename(None, inplace=True)
+
+        event_cond_std_series.loc[cur_event_ids] = cur_model.ranef_df.loc[
+            cur_event_ids, "condsd"
+        ]
+        if site_cname is not None:
+            site_cond_std_series.loc[cur_site_ids] = (
+                cur_model.ranef_df.loc[cur_site_ids, "condsd"]
+                if site_cname is not None
+                else None
+            )
```

Comment:
checking `site_cname` twice is redundant, no?
```suggestion
        if site_cname is not None:
            site_cond_std_series.loc[cur_site_ids] = cur_model.ranef_df.loc[cur_site_ids, "condsd"]
```